Makefile

```diff
@@ -39,15 +26,21 @@ lint-js:
 	jshint ${STATIC_DIR}/js/scribbler-menu.js
 	jshint ${STATIC_DIR}/js/plugins/
 
-build-js:
-	# Build optimized JS
-	# Requires r.js optimizer
-	cd ${STATIC_DIR}/js && r.js -o name=scribbler out=scribbler-min.js baseUrl=. mainConfigFile=scribbler.js
+${STATIC_DIR}/js/bundle.js: ${PROJECT_FILES}
+	node_modules/.bin/browserify $< -o $@
+
+${STATIC_DIR}/js/bundle-min.js: ${STATIC_DIR}/js/bundle.js
+	node_modules/.bin/uglifyjs $^ -o $@
 
-test-js:
+build-js: ${STATIC_DIR}/js/bundle-min.js
+
+${TESTS_DIR}/bundle.js: ${TESTS_DIR}/main.js ${PROJECT_FILES} ${TEST_FILES}
+	node_modules/.bin/browserify -t browserify-compile-templates --extension=.html $^ -o $@
```

Comment:
The tests depend on the project files and you want the tests to be rebuilt if the project files change. It doesn't directly depend on `${STATIC_DIR}/js/bundle.js` because from what I can tell the tests don't use it.